### ISSUE TITLE
modules: Only download what's in config.toml

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#6576]
Right now we download everything inside `go.mod`. This is suboptimal if you have a `big-image-module` you want to avoid getting in "dev mode".

**Comment by bep** [comment]
OK, this was harder than I imagined.

The problem, there seem to be no way to prevent Go to resolve a certain module path without editing the go.mod file itself.

I will add a commit that improves this situation for people using proxies, but to really solve this issue we need to rethink it.

One way would have a way to rewrite the `go.mod` to add replacement directives for disabled modules:

```
replace github.com/big/module => github.com/bep/empty-hugo-module v1.0.0
```

**Comment by bep** [comment]
FIxed in eada236f87d9669885da1ff647672bb3dc6b4954
